manifests/init.pp

```diff
@@ -52,30 +53,31 @@
 #
 class letsencrypt (
   Boolean $configure_epel,
-  Optional[String] $email                = undef,
-  Array $environment                     = [],
-  String $package_name                   = 'certbot',
-  $package_ensure                        = 'installed',
-  String $package_command                = 'certbot',
-  Stdlib::Unixpath $config_dir           = '/etc/letsencrypt',
-  String $config_file                    = "${config_dir}/cli.ini",
-  Hash $config                           = { 'server' => 'https://acme-v02.api.letsencrypt.org/directory' },
-  String $cron_scripts_path              = "${facts['puppet_vardir']}/letsencrypt",
-  String $cron_owner_group               = 'root',
-  Boolean $manage_config                 = true,
-  Boolean $manage_install                = true,
-  Boolean $agree_tos                     = true,
-  Boolean $unsafe_registration           = false,
-  Integer[2048] $key_size                = 4096,
+  Optional[String] $email            = undef,
+  Array $environment                 = [],
+  String $package_name               = 'certbot',
+  $package_ensure                    = 'installed',
+  String $package_command            = 'certbot',
+  Stdlib::Unixpath $config_dir       = '/etc/letsencrypt',
+  String $config_file                = "${config_dir}/cli.ini",
+  Hash $config                       = { 'server' => 'https://acme-v02.api.letsencrypt.org/directory' },
+  String $cron_scripts_path          = "${facts['puppet_vardir']}/letsencrypt",
+  String $cron_owner_group           = 'root',
+  Boolean $manage_config             = true,
+  Boolean $manage_install            = true,
+  Boolean $agree_tos                 = true,
+  Boolean $unsafe_registration       = false,
+  Integer[2048] $key_size            = 4096,
+  Hash[String[1],Hash] $certificates = {},
```

Comment:
Can you add some spec tests for when this parameter is set that it does contain some `letsencrypt::certonly` entries.

Reply:
I'm very new to rspec but: https://github.com/voxpupuli/puppet-letsencrypt/pull/271/commits/9723ce55ffb1410b79b0f0ae4c7776fb53b14ece